src/web/components/App.tsx

```diff
@@ -336,60 +343,88 @@ export const App = ({ CAPI, NAV }: Props) => {
 					pageId={CAPI.pageId}
 				/>
 			</Portal>
-			{CAPI.youtubeMainMediaBlockElement.map((youtubeBlock, index) => (
-				<HydrateOnce
-					key={index}
-					root="youtube-block-main-media"
-					index={youtubeBlock.youtubeIndex}
-				>
-					<YoutubeBlockComponent
-						display={display}
-						design={design}
-						pillar={pillar}
-						hideCaption={false}
-						// eslint-disable-next-line jsx-a11y/aria-role
-						role="inline"
-						adTargeting={adTargeting}
-						isMainMedia={false}
-						id={youtubeBlock.id}
-						assetId={youtubeBlock.assetId}
-						channelId={youtubeBlock.channelId}
-						expired={youtubeBlock.expired}
-						overrideImage={youtubeBlock.overrideImage}
-						posterImage={youtubeBlock.posterImage}
-						duration={youtubeBlock.duration}
-						mediaTitle={youtubeBlock.mediaTitle}
-						altText={youtubeBlock.altText}
-					/>
-				</HydrateOnce>
-			))}
-			{CAPI.youtubeBlockElement.map((youtubeBlock, index) => (
-				<HydrateOnce
-					key={index}
-					root="youtube-block"
-					index={youtubeBlock.youtubeIndex}
-				>
-					<YoutubeBlockComponent
-						display={display}
-						design={design}
-						pillar={pillar}
-						hideCaption={false}
-						// eslint-disable-next-line jsx-a11y/aria-role
-						role="inline"
-						adTargeting={adTargeting}
-						isMainMedia={false}
-						id={youtubeBlock.id}
-						assetId={youtubeBlock.assetId}
-						channelId={youtubeBlock.channelId}
-						expired={youtubeBlock.expired}
-						overrideImage={youtubeBlock.overrideImage}
-						posterImage={youtubeBlock.posterImage}
-						duration={youtubeBlock.duration}
-						mediaTitle={youtubeBlock.mediaTitle}
-						altText={youtubeBlock.altText}
-					/>
-				</HydrateOnce>
-			))}
+			{CAPI.youtubeMainMediaBlockElement.map((youtubeBlock, index) => {
+				// const YoutubeBlockComponentLoadable = loadableDynamicImport(
+				// 	'@frontend/web/components/elements/YoutubeBlockComponent',
+				// 	'YoutubeBlockComponent',
+				// ) as typeof YoutubeBlockComponent;
+				const YoutubeBlockComponent = loadable(
+					() =>
+						import(
+							'@frontend/web/components/elements/YoutubeBlockComponent'
+						),
+					{
+						resolveComponent: (module) =>
+							module.YoutubeBlockComponent,
+					},
+				);
```

Comment:
Hum. Yeah, this is strange, it's in the map as a way to prevent the import being called if it doesn't exist, but maybe it should map over similar to what we do on the server.

Reply:
You mean kind of if (myArray.includes('myThing')) then loadable...?

Reply:
Yeah

Reply:
I think that's okay. It's very expressive code and would be clear from reading it what's going on, esp. if the abstracted way to loadarise a component read well

Reply:
One for after though 👍

Reply:
I've moved this out now if you could take a quick look @oliverlloyd . It still needs abstracting away as there's a bunch of boilerplate but I think this is a good step and the abstraction can be in a separate PR.